lib/workers/repository/process/lookup/filter-checks.ts

```diff
@@ -64,27 +66,82 @@ export async function filterInternalChecks(
         ) {
-          // Skip it if it doesn't pass checks
-          logger.trace(
-            { depName, check: 'minimumReleaseAge' },
-            `Release ${candidateRelease.version} is pending status checks`,
-          );
+          minimumReleaseAgeTimestamp = releaseConfig.minimumReleaseAgeTimestamp;
+        }
+
+        // if there is a releaseTimestamp, regardless of `minimumReleaseAgeTimestamp`, we should process it
+        if (candidateRelease.releaseTimestamp) {
+          // we should skip this if we have a timestamp that isn't passing checks:
+          if (
+            getElapsedMs(candidateRelease.releaseTimestamp) <
+            coerceNumber(toMs(minimumReleaseAge), 0)
+          ) {
+            // Skip it if it doesn't pass checks
+            logger.trace(
+              { depName, check: 'minimumReleaseAge' },
+              `Release ${candidateRelease.version} is pending status checks`,
+            );
+            pendingReleases.unshift(candidateRelease);
+            continue;
+          }
+        } // or if there is no timestamp, and we're running in `minimumReleaseAgeTimestamp=required`
+        else if (
+          is.nullOrUndefined(candidateRelease.releaseTimestamp) &&
+          minimumReleaseAgeTimestamp === 'required'
+        ) {
+          // Skip it, as we require a timestamp
+          candidateVersionsWithoutReleaseTimestamp[
+            minimumReleaseAgeTimestamp
+          ].push(candidateRelease.version);
           pendingReleases.unshift(candidateRelease);
           continue;
+        } // if there is no timestamp, and we're running in `optional` mode, we can allow it
+        else if (
+          is.nullOrUndefined(candidateRelease.releaseTimestamp) &&
+          minimumReleaseAgeTimestamp === 'optional'
+        ) {
+          candidateVersionsWithoutReleaseTimestamp[
+            minimumReleaseAgeTimestamp
+          ].push(candidateRelease.version);
         }
       }
 
+      if (candidateVersionsWithoutReleaseTimestamp.required) {
+        logger.debug(
+          {
+            depName,
+            versions: candidateVersionsWithoutReleaseTimestamp.required,
+            check: 'minimumReleaseAge',
+          },
+          `${candidateVersionsWithoutReleaseTimestamp.required.length} release(s) did not have a releaseTimestamp, and as we're running with minimumReleaseAgeTimestamp=required, these release(s) will be marked as pending status checks`,
+        );
+      } else if (candidateVersionsWithoutReleaseTimestamp.optional) {
+        logger.warn(
+          {
+            depName,
+            versions: candidateVersionsWithoutReleaseTimestamp.optional,
+            check: 'minimumReleaseAge',
+          },
+          `${candidateVersionsWithoutReleaseTimestamp.optional.length} release(s) did not have a releaseTimestamp, but as we're running with minimumReleaseAgeTimestamp=optional, proceeding`,
+        );
+      }
```

Comment:
These are firing incorrectly - should only fire if there are elements in the array